web/src/components/tasks/anytime-action/tax-clearance-certificate/TaxClearanceSteps.tsx

```diff
@@ -1,44 +1,63 @@
 import { HorizontalStepper } from "@/components/njwds-extended/HorizontalStepper";
-import { TaxClearanceStepOne } from "@/components/tasks/anytime-action/tax-clearance-certificate/TaxClearanceStepOne";
-import { TaxClearanceStepThree } from "@/components/tasks/anytime-action/tax-clearance-certificate/TaxClearanceStepThree";
-import { TaxClearanceStepTwo } from "@/components/tasks/anytime-action/tax-clearance-certificate/TaxClearanceStepTwo";
+import { Download } from "@/components/tasks/anytime-action/tax-clearance-certificate/steps/Download";
+import { StepOne } from "@/components/tasks/anytime-action/tax-clearance-certificate/steps/StepOne";
+import { StepThree } from "@/components/tasks/anytime-action/tax-clearance-certificate/steps/StepThree";
+import { StepTwo } from "@/components/tasks/anytime-action/tax-clearance-certificate/steps/StepTwo";
 import { StepperStep } from "@/lib/types/types";
-import { ReactElement } from "react";
+import { ReactElement, useState } from "react";
 
 interface Props {
   steps: StepperStep[];
+  certificatePdfBlob?: Blob;
   currentStep: number;
   stepIndex: (value: ((prevState: number) => number) | number) => void;
   saveTaxClearanceCertificateData: () => void;
   setStepIndex: (step: number) => void;
-  setCertificatePdfBlob: (
-    value: ((prevState: Blob | undefined) => Blob | undefined) | Blob | undefined
-  ) => void;
 }
 
 export const TaxClearanceSteps = (props: Props): ReactElement => {
+  const [certificatePdfBlob, setCertificatePdfBlob] = useState<Blob | undefined>(
+    props.certificatePdfBlob || undefined
+  );
+
   const onStepClick = (step: number): void => {
     if (step === 2 && props.currentStep === 1) {
       props.saveTaxClearanceCertificateData();
     }
     props.setStepIndex(step);
   };
 
-  return (
-    <>
-      <HorizontalStepper steps={props.steps} currentStep={props.currentStep} onStepClicked={onStepClick} />
-      {props.currentStep === 0 && <TaxClearanceStepOne setStepIndex={props.stepIndex} />}
-      {props.currentStep === 1 && (
-        <TaxClearanceStepTwo
+  const steps: { component: ReactElement }[] = [
```

Comment:
I'd like to refine all of these steps a little more and eliminate the need to pass a value to `setStepIndex`, by extracting button logic into it's own function. This would further simplify the step components and make this data structure a little simpler to look at.

Reply:
Opted for the object over an array because there may be other information about the step we would like to index with `props.currentStep`. This gives us that flexibility. If we end up not needing it, this could easily just be turned into an array.